### DISCUSSION TITLE
Fetching initial data globally

### DISCUSSION THREAD

**Discussion by kleinpetr** [nuxt/nuxt#28461] [Questions]
Hi,

time to time I am wondering how we could improve our "fetching" approach to be more optimized and unified.

### codes init 
- for example we have a `countries` store which can be pinia or just function with `useState` variable
- this store is used across the app
#### Possible slutions
1. `initCountries()` function which is called before each usage
1. little workaround when use pinia is to use `useAsyncData(initConutries())` in the store itself
1. use a plugin to call `initCountries()`
1. any other approach?

### shared store between few components
- on the other hand we can have a `blog` store which is used by few components
- this store can be used on the standalone page or within some component on other page to see recent posts
- the store doesn't have to be alive all the time
#### Possible solutions
1. `initBlog()` is called before each usage
2.  little workaround when use pinia is to use `useAsyncData(initConutries())` in the store itself

> I would like to find a unified way how to think about it, also when it comes to SSR.

Thanks for your ideas and thoughts :blush:

**Comment by xjccc** [comment]
Something like this?

[sharedprerenderdata](https://nuxt.com/docs/guide/going-further/experimental-features#sharedprerenderdata)
